### DISCUSSION TITLE
Mac OS resolution issue

### DISCUSSION THREAD

**Discussion by barisyild** [rustdesk/rustdesk#7220] [General]
### Bug Description

Mac remote image quality is very poor.

RustDesk Settings:

Connection method: Direct local ip address
Image Quality: Good image quality
Encoder: VP8, VP9, AV1 are available and all offer similar quality, but I have automatic activated.
Scale: Adaptive

### How to Reproduce

If you connect through windows from a high resolution screen you will notice the problem.

### Expected Behavior

Image quality should be better.

### Operating system(s) on local side and remote side

Windows 11 -> OS X 14.3.1

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

Chrome Remote Desktop:

![image](https://github.com/rustdesk/rustdesk/assets/25794892/1db59ccb-4311-47bc-87a8-ccc68106c8a9)

RustDesk:

![image](https://github.com/rustdesk/rustdesk/assets/25794892/1025054c-45e0-4dee-aa78-2f21e09488cb)

### Additional Context

I think the problem is related to the codec or the scale api of mac os.

As seen in the 1st image, the image resolution is 1512x982 due to scale, but actually this is the scale resolution, the physical image resolution of the device is 3024 x 1964 as in the 2nd image.

Probably based on the resolution in the 2nd image, chrome remote desktop takes the image with a resolution close to the real image on the screen, while rustdesk uses a resolution like 1512x982.

To understand this difference more clearly, try connecting to a macOS device from a high-resolution display.

![image](https://github.com/rustdesk/rustdesk/assets/25794892/7c17637b-897b-4f29-834d-613b71b4e7a9)

![image](https://github.com/rustdesk/rustdesk/assets/25794892/2a1a9436-aab8-487f-8dc3-c491af3680d7)

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/5246
